bliss/simulator/prior.py

```diff
@@ -86,18 +99,35 @@ def __init__(
         self.galaxy_a_scale = galaxy_a_scale
         self.galaxy_a_bd_ratio = galaxy_a_bd_ratio
 
-    def sample_prior(self) -> TileCatalog:
+    def sample_prior(self, batch_rcfs) -> TileCatalog:
         """Samples latent variables from the prior of an astronomical image.
 
+        Args:
+            batch_rcfs: run, camcol, field combinations needed for sampling.
+
         Returns:
             A dictionary of tensors. Each tensor is a particular per-tile quantity; i.e.
             the first three dimensions of each tensor are
             `(batch_size, self.n_tiles_h, self.n_tiles_w)`.
             The remaining dimensions are variable-specific.
         """
         locs = self._sample_locs()
-        galaxy_params = self._sample_galaxy_prior()
-        star_fluxes = self._sample_star_fluxes()
+        select_gal_rcfs = []
+        select_star_rcfs = []
+
+        for rcf in batch_rcfs:
+            rcf = tuple(rcf)
+            if rcf in self.rcf_gals_rest:
+                select_gal_rcfs.append(self.rcf_gals_rest[rcf])  # noqa: WPS529
+            else:  # noqa: WPS529
```

Comment:
Should we disable WPS529 project wide?

Reply:
Aakash and I think that it would make sense to disable it for now since .get() doesn't exactly fit in with the flow of the code in this section. We are also probably going to have to rewrite the majority of this section (and others) anyways when we generalize the decoder to support several other catalogs. I also think it's more readable in this format, personally.